pkg/controllers/subnetbinding/subnetbinding_controller.go

```diff
@@ -270,15 +326,82 @@ func (r *Reconciler) validateVpcSubnetsBySubnetCR(ctx context.Context, namespace
 		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByChildSubnet(subnets[0])
 		if len(bms) > 0 {
 			dependency := r.SubnetBindingService.GetSubnetConnectionBindingMapCRName(bms[0])
-			return nil, &errorWithRetry{
+			return nil, subnetCR, &errorWithRetry{
 				message: fmt.Sprintf("Target Subnet CR %s is associated by %s", name, dependency),
 				error:   fmt.Errorf("target Subnet %s is already associated by SubnetConnectionBindingMap %s", name, dependency),
 				retry:   true,
 			}
 		}
 	}
+	return subnets, subnetCR, nil
+}
 
-	return subnets, nil
+func (r *Reconciler) getSharedSubnetFromNSX(subnet *v1alpha1.Subnet) (*model.VpcSubnet, *errorWithRetry) {
+	anno := subnet.GetAnnotations()
+	associatedResource, ok := anno[servicecommon.AnnotationAssociatedResource]
+	if !ok {
+		return nil, &errorWithRetry{
+			message: fmt.Sprintf("No associated resource annotation in Shared Subnet %s/%s", subnet.Namespace, subnet.Name),
+			error:   fmt.Errorf("no associated resource annotation in Shared Subnet %s/%s", subnet.Namespace, subnet.Name),
+			retry:   false,
+		}
+	}
+	// associatedResource has the format projectID:vpcID:subnetID or :vpcID:subnetID for default project
+	parts := strings.Split(associatedResource, ":")
+	if parts[0] == "" {
+		nc, err := r.VPCService.GetDefaultNetworkConfig()
+		if err != nil {
+			return nil, &errorWithRetry{
+				message: fmt.Sprintf("Failed to get default NetworkConfig, error: %v", err),
+				error:   err,
+				retry:   true,
+			}
+		}
+		_, defaultProject, err := servicecommon.NSXProjectPathToId(nc.Spec.NSXProject)
+		if err != nil {
+			return nil, &errorWithRetry{
+				message: fmt.Sprintf("Failed to parse default NetworkConfig project: %s", nc.Spec.NSXProject),
+				error:   err,
+				retry:   true,
+			}
+		}
+		parts[0] = defaultProject
+	}
+	// Get Subnet path from NetworkConfig
+	nc, err := r.VPCService.GetVPCNetworkConfigByNamespace(subnet.Namespace)
+	if err != nil {
+		return nil, &errorWithRetry{
+			message: fmt.Sprintf("Failed to get NetworkConfig for Namespace %s", subnet.Namespace),
+			error:   err,
+			retry:   true,
+		}
+	}
+	for _, subnetPath := range nc.Spec.Subnets {
+		vpcInfo, err := servicecommon.ParseVPCResourcePath(subnetPath)
+		if err != nil {
+			return nil, &errorWithRetry{
+				message: fmt.Sprintf("Failed to parse path: %s", subnetPath),
+				error:   err,
+				retry:   false,
+			}
+		}
+		if parts[0] == vpcInfo.ProjectID && parts[1] == vpcInfo.VPCID && parts[2] == vpcInfo.ID {
+			nsxSubnet, err := r.SubnetService.GetSubnetFromNSX(vpcInfo)
+			if err != nil {
+				return nil, &errorWithRetry{
+					message: fmt.Sprintf("Failed to get Subnet from NSX, error: %v", err),
+					error:   err,
+					retry:   true,
+				}
+			}
+			return nsxSubnet, nil
```

Comment:
Do you mean only one Subnet in a single VPC (`nc.Spec.Subnets`) can be shared?

Reply:
This part of code is removed. But I think the answer for this question is as we are looking for the child subnet referred by bindingmap, it is fine to return the nsx Subnet once we find one match the path.